stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/compiler/SocketCompilerPlugin.java

```diff
@@ -0,0 +1,196 @@
+    public void init(DiagnosticLog diagnosticLog) {
+        this.diagnosticLog = diagnosticLog;
+    }
+
+    @Override
+    public void process(EndpointNode endpointNode, List<AnnotationAttachmentNode> annotations) {
+        final ExpressionNode configurationExpression = endpointNode.getConfigurationExpression();
+        if (NodeKind.RECORD_LITERAL_EXPR.equals(configurationExpression.getKind())) {
+            BLangRecordLiteral recordLiteral = (BLangRecordLiteral) configurationExpression;
+            boolean isPortPresent = false;
+            for (BLangRecordLiteral.BLangRecordKeyValue config : recordLiteral.getKeyValuePairs()) {
+                final String key = ((BLangSimpleVarRef) config.getKey()).variableName.value;
+                if (CONFIG_FIELD_PORT.equals(key)) {
+                    isPortPresent = true;
+                    final Object value = ((BLangLiteral) config.getValue()).getValue();
+                    Long port = (Long) value;
+                    if (port <= 0) {
+                        String msg = String.format("'%s' must be a positive integer value", CONFIG_FIELD_PORT);
+                        diagnosticLog.logDiagnostic(ERROR, endpointNode.getPosition(), msg);
+                        break;
+                    }
+                }
+            }
+            if (!isPortPresent) {
+                String msg = String.format("'%s' is a mandatory configuration field", CONFIG_FIELD_PORT);
+                diagnosticLog.logDiagnostic(ERROR, endpointNode.getPosition(), msg);
+            }
+        }
+    }
+
+    @Override
+    public void process(ServiceNode serviceNode, List<AnnotationAttachmentNode> annotations) {
+        List<BLangResource> resources = (List<BLangResource>) serviceNode.getResources();
+        resources.forEach(res -> validate(serviceNode.getName().getValue(), res, this.diagnosticLog));
+    }
+
+    private void validate(String serviceName, BLangResource resource, DiagnosticLog diagnosticLog) {
+        switch (resource.getName().getValue()) {
+            case RESOURCE_ON_CONNECT:
+            case RESOURCE_ON_ACCEPT:
+                validateOnAccept(serviceName, resource, diagnosticLog);
+                break;
+            case RESOURCE_ON_READ_READY:
+                validateOnReadReady(serviceName, resource, diagnosticLog);
+                break;
+            case RESOURCE_ON_CLOSE:
+                validateOnClose(serviceName, resource, diagnosticLog);
+                break;
+            case RESOURCE_ON_ERROR:
+                validateOnError(serviceName, resource, diagnosticLog);
+                break;
+            default:
+        }
+    }
+
+    private void validateOnError(String serviceName, BLangResource resource, DiagnosticLog diagnosticLog) {
+        final List<BLangSimpleVariable> readReadyParams = resource.getParameters();
+        if (readReadyParams.size() != 2) {
+            String msg = String.format("Invalid resource signature for %s in service %s. "
```

Comment:
Can't we define a constant for this part `Invalid resource signature for %s in service %s.` Same thing is used in several other places.